contracts/optimistic-ethereum/OVM/verification/OVM_StateTransitioner.sol

```diff
@@ -353,8 +344,13 @@ contract OVM_StateTransitioner is Lib_AddressResolver, OVM_FraudContributor, iOV
         contributesToFraudProof(preStateRoot, transactionHash)
     {
         require(
+            ovmStateManager.getTotalUncommittedContractStorage() == 0,
```

Comment:
Force users to prove all storage slots first so that we know the `storageRoot` values are finalized.